src/views/CardStack.js

```diff
@@ -191,49 +192,42 @@ class CardStack extends React.Component<DefaultProps, Props, void> {
     return transitionSpec;
   }
 
-  _renderHeader(props: NavigationTransitionProps, headerMode: HeaderMode): ?React.Element<*> {
-    const navigation = this._getChildNavigation(props.scene);
-    const header = this.props.router.getScreenConfig(navigation, 'header') || {};
+  _renderHeader(
+    transitionProps: NavigationTransitionProps,
+    headerMode: HeaderMode
+  ): ?React.Element<*> {
+    const headerConfig = this.props.router.getScreenConfig(
+      transitionProps.navigation,
+      'header'
+    ) || {};
 
     return (
       <this.props.headerComponent
-        {...props}
-        style={header.style}
+        {...transitionProps}
+        router={this.props.router}
+        style={headerConfig.style}
         mode={headerMode}
         onNavigateBack={() => this.props.navigation.goBack(null)}
-        renderLeftComponent={(props) => {
-          const navigation = this._getChildNavigation(props.scene);
-          const header = this.props.router.getScreenConfig(navigation, 'header');
+        renderLeftComponent={(props: NavigationTransitionProps) => {
+          const header = this.props.router.getScreenConfig(props.navigation, 'header');
           if (header && header.left) {
             return header.left;
           }
-          const { renderLeftComponent } = this.props.headerComponent.defaultProps || {};
-          if (typeof renderLeftComponent === 'function') {
-            return renderLeftComponent(props);
-          }
-          return null;
+          return undefined;
         }}
-        renderRightComponent={({ scene }) => {
-          const navigation = this._getChildNavigation(scene);
-          const header = this.props.router.getScreenConfig(navigation, 'header');
+        renderRightComponent={(props: NavigationTransitionProps) => {
+          const header = this.props.router.getScreenConfig(props.navigation, 'header');
           if (header && header.right) {
             return header.right;
           }
-          return null;
+          return undefined;
         }}
-        renderTitleComponent={({ scene }) => {
-          const navigation = this._getChildNavigation(scene);
-          const header = this.props.router.getScreenConfig(navigation, 'header');
-          let title = null;
-          if (header && header.title) {
-            title = header.title;
-          } else {
-            title = this.props.router.getScreenConfig(navigation, 'title');
-          }
-          if (typeof title === 'string') {
-            return <Header.Title>{title}</Header.Title>;
+        renderTitleComponent={(props: NavigationTransitionProps) => {
+          const header = this.props.router.getScreenConfig(props.navigation, 'header');
+          if (header && header.title && React.isValidElement(header.title)) {
```

Comment:
This also needs to support `null`

Check for `(header && typeof header.title !== 'undefined' && (React.isValidElement(header.title) || header.title === null))`